riscv/interactive.cc

```diff
@@ -159,17 +159,28 @@ void sim_t::interactive()
 
   while (!done())
   {
-    std::cerr << ": " << std::flush;
-    std::string s = readline(2);
+#ifdef HAVE_BOOST
+    streambuf bout; // socket output
+#endif
+    string s;
+#ifdef HAVE_BOOST
+       s = rin(&bout); // get comand string from socket or terminal
+#else
+       cerr << ": " << std::flush;
+       s = readline(2); // 2 is stderr which in turn is stdin
+#endif
```

Comment:
If you move the declaration of `string s` up then you'll only need one `#ifdef` in this block

Reply:
fixed

Reply:
fixed in original commit